src/rules/no-invalid-escape-sequences.ts

```diff
@@ -26,8 +27,8 @@ const rule: Rule.RuleModule = {
   },
 
   create(context): Rule.RuleListener {
-    // variables should be defined here
-    const escapePattern = /(^|[^\\])\\([1-7][0-7]*|[0-7]{2,})+/;
+    const source = context.getSourceCode();
+    const escapePattern = /((^|[^\\])\\([1-7][0-7]*|[0-7]{2,}))/g;
```

Comment:
Might be better to use a regex like this that matches odd numbers of `\`

```suggestion
    const escapePattern = /(^\\(\\\\)*([1-7][0-7]*|[0-7]{2,}))/g;
```

This will match `\0123` as well as `\\\0123` but not `\\0123`